features/exploredash/src/main/java/org/dash/wallet/features/exploredash/repository/DashDirectRepository.kt

```diff
@@ -34,22 +34,67 @@ class DashDirectRepository @Inject constructor(
     private val authApi: DashDirectAuthApi,
     private val config: DashDirectConfig
 ) : DashDirectRepositoryInt {
-
     override suspend fun signIn(
-        email: String,
-        password: String
+        email: String
+    ): ResponseResource<Boolean> = safeApiCall {
+        authApi.signIn(email = email)
+            .also {
+                it?.errorMessage?.let { errorMessage ->
+                    if (errorMessage.isNotEmpty()) {
+                        throw Exception(errorMessage)
+                    }
+                }
+                if(it?.data?.statusCode==0){
+                    createUser(email)
+                }
+                config.setPreference(DashDirectConfig.PREFS_KEY_DASH_DIRECT_EMAIL, email)
+            }
+        true
+    }
+
+    override suspend fun createUser(email: String): ResponseResource<Boolean> = safeApiCall {
+        authApi.createUser(email = email)
+            .also {
+                it?.data?.errorMessage?.let { errorMessage ->
+                    if (errorMessage.isNotEmpty()) {
+                        throw Exception(errorMessage)
+                    }
+                }
+                config.setPreference(DashDirectConfig.PREFS_KEY_DASH_DIRECT_EMAIL, email)
+            }
+        true
+    }
+
+    override suspend fun verifyEmail(
+        code: String
     ): ResponseResource<Boolean> = safeApiCall {
-        authApi.signIn(signInRequest = SignInRequest(emailAddress = email, password = password))
+        authApi.verifyEmail(signInRequest = VerifyEmailRequest(emailAddress = config.getPreference(DashDirectConfig.PREFS_KEY_DASH_DIRECT_EMAIL), code = code))
             .also {
-                it?.data?.token?.let { token ->
+                it?.data?.errorMessage?.let { errorMessage ->
+                    if (it?.data?.hasError == true && errorMessage.isNotEmpty()) {
+                        throw Exception(errorMessage)
+                    }
+                }
+                it?.data?.accessToken?.let { token ->
                     config.setPreference(DashDirectConfig.PREFS_KEY_LAST_DASH_DIRECT_ACCESS_TOKEN, token)
                 }
             }
         config.getPreference(DashDirectConfig.PREFS_KEY_LAST_DASH_DIRECT_ACCESS_TOKEN)?.isNotEmpty() ?: false
     }
 
     override fun isUserSignIn() =
-        runBlocking { config.getPreference(DashDirectConfig.PREFS_KEY_LAST_DASH_DIRECT_ACCESS_TOKEN)?.isNotEmpty() ?: true }
+        runBlocking { config.getPreference(DashDirectConfig.PREFS_KEY_LAST_DASH_DIRECT_ACCESS_TOKEN)?.isNotEmpty() ?: false }
+
+    override fun getDashDirectEmail(): String? {
+        return runBlocking { config.getPreference(DashDirectConfig.PREFS_KEY_DASH_DIRECT_EMAIL) }
+    }
+
+    override fun logout() {
+        runBlocking {
+            config.setPreference(DashDirectConfig.PREFS_KEY_LAST_DASH_DIRECT_ACCESS_TOKEN, "")
+            config.setPreference(DashDirectConfig.PREFS_KEY_DASH_DIRECT_EMAIL, "")
+        }
+    }
```

Comment:
As I've mentioned before, the use of `runBlocking` is discouraged. Reading the preferences is an asynchronious I/O operation and should be treated as such. On slow devices, these calls can cause ANRs.

Please remove `runBlocking`, make the methods `suspend`, and call them from a coroutine context.